storage/src/backends/in_memory.rs

```diff
@@ -12,28 +12,28 @@ use std::collections::HashMap;
 #[derive(Debug, Eq, PartialEq)]
 pub struct InMemoryStorage<'a> {
     /// A HashMap to implement easy and fast K/V lookup
-    pub memory: HashMap<&'a [u8], Vec<u8>>
+    pub memory: HashMap<&'a [u8], Vec<u8>>,
 }
 
 /// Implement the Storage generic trait for the InMemoryStorage storage data structure.
-impl <'a> Storage<(), &'a [u8], Vec<u8>> for InMemoryStorage<'a> {
-
-    fn new(_: ()) -> Result<Box<Self>> {
-        Ok(Box::new(InMemoryStorage { memory: HashMap::new() }))
+impl<'a> Storage<&'a [u8], Vec<u8>> for InMemoryStorage<'a> {
+    fn new(_: String) -> StorageResult<Box<Self>> {
```

Comment:
Why use `String` here as input type instead of `()`? We don't even want to pass any parameters to `InMemoryStorage::new()`. Am I missing some 2018 novelty?